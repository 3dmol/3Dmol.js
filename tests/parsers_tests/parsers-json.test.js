
global.$ = require("jquery");
global.URL.createObjectURL = function() {};
let $3Dmol = require("../../build/3Dmol.js");
const fs = require('fs');

describe('Function json |', ()=>{
    const data = '{"m":[{"a":[{"x":85,"y":144},{"x":102.32050807568878,"y":134},{"x":67.67949192431124,"y":134},{"x":85,"y":164}],"b":[{"b":0,"e":1},{"b":0,"e":2},{"b":0,"e":3}]},{"a":[{"x":213,"y":131},{"x":230.32050807568876,"y":121},{"x":247.64101615137753,"y":131},{"x":264.9615242270663,"y":121}],"b":[{"b":0,"e":1},{"b":1,"e":2},{"b":2,"e":3}]}]}';
    let atoms = $3Dmol.Parsers.json(data, {});

    test("Atoms should match the snapshot", ()=>{
        expect(atoms).toMatchSnapshot();
    });

});
