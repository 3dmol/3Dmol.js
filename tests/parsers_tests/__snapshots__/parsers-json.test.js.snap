// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Function json | Atoms should match the snapshot 1`] = `
[
  [
    {
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        1,
        2,
        3,
      ],
      "elem": "C",
      "id": undefined,
      "serial": 0,
      "x": 85,
      "y": 144,
      "z": 0,
    },
    {
      "bondOrder": [
        1,
      ],
      "bonds": [
        0,
      ],
      "elem": "C",
      "id": undefined,
      "serial": 1,
      "x": 102.32050807568878,
      "y": 134,
      "z": 0,
    },
    {
      "bondOrder": [
        1,
      ],
      "bonds": [
        0,
      ],
      "elem": "C",
      "id": undefined,
      "serial": 2,
      "x": 67.67949192431124,
      "y": 134,
      "z": 0,
    },
    {
      "bondOrder": [
        1,
      ],
      "bonds": [
        0,
      ],
      "elem": "C",
      "id": undefined,
      "serial": 3,
      "x": 85,
      "y": 164,
      "z": 0,
    },
  ],
]
`;
